Podcast/SettingsPageViewController.swift

```diff
@@ -0,0 +1,207 @@
+// String value is cell reuse ID
+enum SettingsFieldType {
+    case button
+    case textField
+    case disclosure
+    case label
+    case toggle
+    
+    static let allValues: [SettingsFieldType] = [.button, .textField, .disclosure, .label, .toggle]
+    
+    var reuseIdAndClass: (String, AnyClass) {
+        switch self {
+        case .textField:
+            return ("textFieldSettingsCell", TextFieldSettingsTableViewCell.self)
+        default:
+            return ("settingsCell", SettingsTableViewCell.self)
+        }
+    }
+}
+
+/**
+ * Create a section in settings by setting these
+ */
+struct SettingsSection {
+    var header: String?
+    var footer: String?
+    var items: [SettingsField]
+    
+    init(items: [SettingsField]) {
+        self.items = items
+    }
+}
+
+/**
+ * Each on of these in the items array of a section will be shown in that section
+ */
+struct SettingsField {
+    var title: String
+    var saveFunction: ((Any) -> Void)
+    var type: SettingsFieldType
+    var placeholder: String // Only used for TextField!
+    var tapAction: (() -> Void) // User for disclosure, but applies to all
+    
+    init(title: String, saveFunction: @escaping ((Any) -> Void) = {_ in }, type: SettingsFieldType = .label, placeholder: String = "", tapAction: @escaping (() -> Void) = {}) {
+        self.title = title
+        self.saveFunction = saveFunction
+        self.type = type
+        self.placeholder = placeholder
+        self.tapAction = tapAction
+    }
+}
+
+/**
+ * SettingsPageViewController is a settings controller where changing the sections and items
+ * array will dynamically create a settings view with desired actions.
+ */
+class SettingsPageViewController: ViewController, UITableViewDelegate, UITableViewDataSource {
+
+    // View items
+    private var tableView: UITableView!
+    private var saveButton: UIBarButtonItem!
+    
+    // Can change this but not recommended
+    var sectionSpacing: CGFloat = 18
+    
+    // Settable configurations
+    var sections: [SettingsSection] = []
+    var showSave: Bool! {
+        didSet {
+            if showSave {
+                navigationItem.rightBarButtonItem = saveButton
+            } else {
+                navigationItem.rightBarButtonItem = nil
+            }
+        }
+    }
+    var returnOnSave:Bool = true
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        navigationController?.setNavigationBarHidden(false, animated: false)
+        
+        tableView = UITableView(frame: CGRect.zero, style: .grouped)
+        tableView.showsVerticalScrollIndicator = true
+        tableView.allowsSelection = false
+        tableView.backgroundColor = .paleGrey
+        for type in SettingsFieldType.allValues {
+            let (reuseId, cellClass):(String, AnyClass) = type.reuseIdAndClass
+            tableView.register(cellClass, forCellReuseIdentifier: reuseId)
+        }
+        tableView.separatorInset = UIEdgeInsetsMake(0, 0, 0, 0)
```

Comment:
Do we need this? isn't the default to have no edge insets

Reply:
No the default I believe has like a left of 15 I think...